SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/StaticSubobjectsTest/StaticSubobjectsTest.cpp

```diff
@@ -0,0 +1,353 @@
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents - 1,
+						TEXT("AStaticSubobjectTestActor should have one less than the initial number of components on the server."));
+		DestroyOneNonRootComponent();
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents - 2,
+						TEXT("AStaticSubobjectTestActor should have had another component removed"));
+		FinishStep();
+	});
+
+	// Step 14
+	CheckClientNumberComponentsOnTestActorWithWait(InitialNumComponents);
+
+	// Step 15
+	MoveClientPawn(PawnRemoteLocation);
+
+	// Step 16
+	CheckClientSeeIntProperty(2);
+
+	// Step 17
+	CheckClientNumberComponentsOnTestActorWithWait(InitialNumComponents - 1);
+
+	// Step 18 - Server Cleanup.
+	AddStep(TEXT("StaticSubobjectsTestServerCleanup"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		// Possess the original pawn, so that the spawned character can get destroyed correctly
+		ASpatialFunctionalTestFlowController* ClientOneFlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		APlayerController* PlayerController = Cast<APlayerController>(ClientOneFlowController->GetOwner());
+
+		if (AssertIsValid(PlayerController, TEXT("PlayerController should be valid")))
+		{
+			PlayerController->Possess(ClientDefaultPawn);
+			FinishStep();
+		}
+	});
+}
+
+void AStaticSubobjectsTest::MoveClientPawn(FVector& ToLocation)
+{
+	// Server moves Client 1's pawn to the location
+	AddStep(TEXT("StaticSubobjectsTestServerMoveClient"), FWorkerDefinition::Server(1), nullptr, [this, &ToLocation]() {
+		if (AssertIsValid(ClientSpawnedPawn, TEXT("ClientSpawnedPawn should be valid")))
+		{
+			ClientSpawnedPawn->SetActorLocation(ToLocation);
+			AssertEqual_Vector(ClientSpawnedPawn->GetActorLocation(), ToLocation, TEXT("Client pawn should have been moved to location"),
+							   1.0f);
+		}
+		FinishStep();
+	});
+
+	// Client 1 makes sure that the movement was correctly replicated
+	AddStep(
+		TEXT("StaticSubobjectsTestClientCheckMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this, &ToLocation](float DeltaTime) {
+			APawn* PlayerCharacter = GetFlowPawn();
+
+			if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid")))
+			{
+				RequireEqual_Vector(PlayerCharacter->GetActorLocation(), ToLocation, TEXT("Character was not moved to location"), 1.0f);
+				FinishStep();
+			}
+		},
+		StepTimeLimit);
+}
+
+void AStaticSubobjectsTest::CheckClientCanNotSeeIntPropertyWithWait(int ShouldntSeeVal)
+{
+	AddStep(
+		TEXT("StaticSubobjectsTestClientCheckIntValueDidntIncreaseWithWait"), FWorkerDefinition::Client(1), nullptr,
+		[this]() {
+			StepTimer = 0.f;
+		},
+		[this, ShouldntSeeVal](const float DeltaTime) {
+			RequireNotEqual_Int(TestActor->TestIntProperty, ShouldntSeeVal,
+								TEXT("The updated TestIntProperty shouldn't have been replicated"));
+			StepTimer += DeltaTime;
+			if (StepTimer >= 0.5f)
```

Comment:
This constant might be better specified not inline, the same as StepTimeLimit; this important parameter is kinda hidden inside of this method.

Reply:
yee, moved it out